engine/source/public/core/fileio.hpp

```diff
@@ -0,0 +1,47 @@
+﻿#pragma once
+#include <map>
+#include <string_view>
+#include <vector>
+
+namespace lumina
+{
+    class Engine;
+
+    class FileIO
+    {
+    public:
+        FileIO();
+        ~FileIO() = default;
+
+        FileIO(const FileIO&)            = delete;
+        FileIO(FileIO&&)                 = delete;
+        FileIO& operator=(const FileIO&) = delete;
+        FileIO& operator=(FileIO&&)      = delete;
+
+        enum class Directory
+        {
+            Assets,
+            Config,
+            Log
+        };
+
+        //Text file read/write
+        std::string ReadTextFile(Directory directory, const std::string& filePath);
```

Comment:
We could add attributes, in this case we could for example use `[[nodiscard]]` since we always want to use the return value when this function is called

Reply:
How do you see this then?

Reply:
Like this, since when you call the `ReadTextFile` you always want to use it's return value

```suggestion
        [[nodiscard]] std::string ReadTextFile(Directory directory, const std::string& filePath);
```